src/routes/committee/setlist/+page.svelte

```diff
@@ -0,0 +1,249 @@
+		musiciansForSongs: Map<string, [string, string][]>
+		users: user[]
+		namesMap: { [userId: string]: { participantName: string } }
+	}
+
+	let selectedSong: number
+	let selectedUserEntry: number
+	let selectedUser: string
+	let selectedInstrument: string
+	let openDel = false
+	let openDelUser = false
+	let openAdd = false
+
+	let participantListItems: DropdownItem[] = data.users.map(
+		(user) => ({ id: user.id, text: user.name } as DropdownItem)
+	)
+
+	let participant: string
+	let instrument: string
+
+	async function RemoveSongFromSetlist() {
+		// TODO: use current edition
+		const editionRef = doc(db, 'editions', 'ZI3Eab1mXjHvCUS47o40')
+		updateDoc(editionRef, {
+			songs: arrayRemove(doc(db, 'songs', data.songs[selectedSong].id))
+		})
+		data.songs.splice(selectedSong, 1)
+		data.songs = data.songs
+	}
+
+	async function AddParticipantToSong() {
+		await addDoc(collection(db, 'users/' + participant + '/playsSongInEdition'), {
+			edition: doc(db, 'editions', 'ZI3Eab1mXjHvCUS47o40'),
+			part: instrument,
+			song: doc(db, 'songs', data.songs[selectedSong].id)
+		})
+		data.musiciansForSongs[data.songs[selectedSong].id] = [
+			...data.musiciansForSongs[data.songs[selectedSong].id],
+			[data.namesMap[participant], instrument]
+		]
+	}
+
+	async function RemoveParticipantFromSong() {
+		const userQuery = query(collection(db, 'users'), where('name', '==', selectedUser))
+		const userId = (await getDocs(userQuery)).docs[0].id
+		const songRef = doc(db, 'songs', data.songs[selectedSong].id)
+
+		const playsInQuery = query(
+			collection(db, 'users', userId, 'playsSongInEdition'),
+			where('song', '==', songRef),
+			where('part', '==', selectedInstrument)
+		)
+		const playsInDoc = (await getDocs(playsInQuery)).docs[0].ref
+
+		await deleteDoc(playsInDoc)
+
+		data.musiciansForSongs = data.musiciansForSongs
+	}
+</script>
+
+<Grid>
+	<Row>
+		<Column><h1 class="titleText">Setlist</h1></Column>
+	</Row>
+
+	<StructuredList condensed>
+		<StructuredListHead>
+			<StructuredListRow head>
+				<StructuredListCell head>Title</StructuredListCell>
+				<StructuredListCell head>Artist</StructuredListCell>
+				<StructuredListCell head>Link</StructuredListCell>
+				<StructuredListCell head>Line-up</StructuredListCell>
+			</StructuredListRow>
+		</StructuredListHead>
+		{#each data.songs as song, i}
+			<StructuredListRow>
+				<StructuredListCell>{song.name}</StructuredListCell>
+				<StructuredListCell>{song.artist}</StructuredListCell>
+				<StructuredListCell
+					>{@const validUrl = isValidUrl(song.link)}
+					<Button
+						href={validUrl ? song.link : undefined}
+						target="blank"
+						kind="ghost"
+						size="small"
+						iconDescription={validUrl ? song.link : 'Invalid URL'}
+						icon={PlayFilledAlt}
+						disabled={!validUrl}
+					/></StructuredListCell
+				>
+				<StructuredListCell>
+					{#each Object.entries(data.musiciansForSongs) as [key, value], j}
+						{#if key == song.id}
```

Comment:
`{@const musicians = data.musiciansForSongs[song.id]}` zou moeten werken, daar heb je de dict voor :)

Reply:
of iets in die buurt